### PR TITLE
Implement UnusedArgument cop

### LINKED ISSUE #53: New cop: UnusedArgument
Arguments should either be listed explicitly or `**rest` should be in the `resolve` signature

```ruby
# good
class SomeResolver < Resolvers::Base
  type SomeType, null: false

  argument :arg1, String, required: true
  argument :arg2, String, required: true

  def resolve(arg1:, arg2:); end
end

class SomeResolver < Resolvers::Base
  type SomeType, null: false

  argument :arg1, String, required: true
  argument :arg2, String, required: true

  def resolve(arg1:, **rest); end
end

class SomeResolver < Resolvers::Base
  type SomeType, null: false

  argument :arg1, String, required: true
  argument :arg2, String, required: true

  def resolve(args); end
end

# bad
class SomeResolver < Resolvers::Base
  type SomeType, null: false

  argument :arg1, String, required: true
  argument :arg2, String, required: true

  def resolve(arg1:); end
end
```

### PR DESCRIPTION
**Pull request by aishek**
First of all, I've implemented autocorrection, and I am not sure if it is better than just add errors to arguments.

And the second thing is the cop ignores the case when there is no `resolve` method at all.

What do you think?

Resolves #53

### CODE REVIEW

**Review by DmitryTsepelev — CHANGES_REQUESTED** [review]
Good start! Please check out my comments below and let's make CI pass

**Comment by aishek** [comment]
Thanks for the comments! I've applied them, and also made the code backward compatible.

I am not sure if the following method if reliable enough:

```ruby
def method_name(node)
  node.loc.keyword.end.resize(node.method_name.to_s.size + 1)
end
```

`Util.args_begin` won't work for me somehow.

File: lib/rubocop/cop/graphql/unused_argument.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          if unresolved_args.any?
+            register_offense(resolve_method_node, unresolved_args)
+          end
+        end
+
+        private
+
+        def find_resolve_method_node(node)
+          resolve_method_nodes = resolve_method_definition(node)
+          return unless resolve_method_nodes.any?
```
[the remark this answers is not in the document]
Comment (DmitryTsepelev) [reply lib/rubocop/cop/graphql/unused_argument.rb]: well, that was a bad suggestion, specs failing because of it 🥲

**Comment by DmitryTsepelev** [comment]
Specs are still failing (because of my stupid suggestion mostly) and we have a number of rubocop offences 🙂

**Comment by aishek** [comment]
Ok, I fixed it all :crossed_fingers:

**Review by DmitryTsepelev — APPROVED** [review]
LGTM, thank you!

**Comment by jgrau** [comment]
Was the `loads:` argument considered? I have this chunk of code in a mutation:

```
...
    argument :booking_id, String, loads: Types::BookingType

    def resolve(booking:)
      ...
    end
...
```

but that complains that it wants `booking_id` to be a keyword argument in the `resolve` method.

:)

**Comment by aishek** [comment]
> Was the `loads:` argument considered?

No, it was not. I will fix it, thanks!

**Comment by aishek** [comment]
I've created the fix https://github.com/DmitryTsepelev/rubocop-graphql/pull/65

**Pull request merged**